Model/src/main/java/Action/IdFactory.java

```diff
@@ -1,14 +1,17 @@
 package Action;
 
-import DataBase.DataBaseInf;
-import Users.Role;
+import java.util.concurrent.atomic.AtomicInteger;
 
-public class IdFactory {
-    private static int generalID =0;
-
-    public static int idBuilder () {
-        generalID ++;
-
-        return generalID;
+/**
+This class factory individual id for every
+essence in database
+**/
+public final class IdFactory {
+    private IdFactory() {
+    }
+    private static volatile int generalID = 0;
+    public static synchronized int  idBuilder() {
+        AtomicInteger id =  new AtomicInteger(generalID++);
+        return id.get();
     }
```

Comment:
Странное использование атомика. Atomic, как раз таки, предназначен для работы в многопоточке. Синхронизация тут не нужна. И нет смысла каждый раз создавать новый объект для него, только перерасход памяти. Попробуй что-нибудь типа такого:
```suggestion
    public static final AtomicInteger generalID = new AtomicInteger(0);
    
    public static int nextId() {
        return generalID.incrementAndGet();
    }
```

Reply:
Генератор переделан.